examples/platform/silabs/efr32/rs911x/hal/efx_spi.c

```diff
@@ -141,13 +158,131 @@ void rsi_hal_board_init(void)
+#endif /* EFR32MG24 */
+
     /* GPIO INIT of MG12 & MG24 : Reset, Wakeup, Interrupt */
     sl_wfx_host_gpio_init();
 
     /* Reset of Wifi chip */
     sl_wfx_host_reset_chip();
+    SILABS_LOG("%s completed.", __func__);
+}
+
+#if defined(EFR32MG24)
+
+void SPIDRV_ReInit(uint32_t baudrate)
+{
+    if (EUSART_BaudrateGet(MY_USART) == baudrate)
+    {
+        // EUSART synced to baudrate already
+        return;
+    }
+    EUSART_BaudrateSet(MY_USART, 0, baudrate);
+}
+
+sl_status_t sl_wfx_host_spi_cs_assert(void)
+{
+    configASSERT(spi_sem_sync_hdl);
+    if (xSemaphoreTake(spi_sem_sync_hdl, portMAX_DELAY) != pdTRUE)
+    {
+        return SL_STATUS_TIMEOUT;
+    }
+
+    if (!spi_enabled) // Reduce SPIDRV_Init
+    {
+        SPIDRV_Init(SL_SPIDRV_HANDLE, &sl_spidrv_eusart_init_exp);
+        spi_enabled = true;
+    }
+    GPIO_PinOutClear(SL_SPIDRV_EUSART_EXP_CS_PORT, SL_SPIDRV_EUSART_EXP_CS_PIN);
+    return SL_STATUS_OK;
+}
+
+sl_status_t sl_wfx_host_spi_cs_deassert(void)
+{
+    if (spi_enabled)
+    {
+        SPIDRV_DeInit(SL_SPIDRV_HANDLE);
+        spi_enabled = false;
+    }
+    GPIO_PinOutSet(SL_SPIDRV_EUSART_EXP_CS_PORT, SL_SPIDRV_EUSART_EXP_CS_PIN);
+    GPIO->EUSARTROUTE[SL_SPIDRV_EUSART_EXP_PERIPHERAL_NO].ROUTEEN = 0;
+    xSemaphoreGive(spi_sem_sync_hdl);
+    return SL_STATUS_OK;
+}
+
+void sl_wfx_host_spiflash_cs_assert(void)
+{
+    GPIO_PinOutClear(SL_MX25_FLASH_SHUTDOWN_CS_PORT, SL_MX25_FLASH_SHUTDOWN_CS_PIN);
 }
 
+void sl_wfx_host_spiflash_cs_deassert(void)
+{
+    GPIO_PinOutSet(SL_MX25_FLASH_SHUTDOWN_CS_PORT, SL_MX25_FLASH_SHUTDOWN_CS_PIN);
+}
+
+void sl_wfx_host_pre_bootloader_spi_transfer(void)
+{
+    if (xSemaphoreTake(spi_sem_sync_hdl, portMAX_DELAY) != pdTRUE)
+    {
+        return;
+    }
+    if (spi_enabled)
+    {
+        SPIDRV_DeInit(SL_SPIDRV_HANDLE);
+        spi_enabled = false;
+    }
+    int32_t status = bootloader_init();
+    // bootloader_init takes care of SPIDRV_Init()
+    if (status != BOOTLOADER_OK)
+    {
+        SILABS_LOG("%s: bootloader_init failed with error code: %d", __func__, status);
+        return;
+    }
+    sl_wfx_host_spiflash_cs_assert();
+}
+
+void sl_wfx_host_post_bootloader_spi_transfer(void)
+{
+    int32_t status = bootloader_deinit();
+    // bootloader_deinit will do USART disable
+    if (status != BOOTLOADER_OK)
+    {
+        SILABS_LOG("%s: bootloader_deinit failed with error code: %d", __func__, status);
+        return;
+    }
+    GPIO->USARTROUTE[SL_MX25_FLASH_SHUTDOWN_PERIPHERAL_NO].ROUTEEN = 0;
+    sl_wfx_host_spiflash_cs_deassert();
+    xSemaphoreGive(spi_sem_sync_hdl);
+}
+
+void sl_wfx_host_pre_lcd_spi_transfer(void)
+{
+    if (xSemaphoreTake(spi_sem_sync_hdl, portMAX_DELAY) != pdTRUE)
+    {
+        return;
+    }
+    if (spi_enabled)
+    {
+        SPIDRV_DeInit(SL_SPIDRV_HANDLE);
+        spi_enabled = false;
+    }
+    sl_memlcd_refresh(sl_memlcd_get());
+    // sl_memlcd_refresh takes care of SPIDRV_Init()
```

Comment:
Why is this line commented out and not removed ? If we need to keep it, a comment would be nice